Validate data

FelipeCybis opened the pull request:
Validating data with hash files where we test the actual SHA1 hash for all data in a directory to the expected hash saved in a .txt file.
obs.: the PR changing a little bit the logic of this function was made some of the commits obsolete and generated some merge conflicts on this PR, but I assume now it is all ok. The main file changes is `validate_data.py`

FelipeCybis committed:
Added hash list file

FelipeCybis committed:
Imported only `sha1` from `hashlib`

FelipeCybis committed:
Added function that returns hash from filename
* Get byte contents of file `filename`, return SHA1 hash

FelipeCybis committed:
Improved docstring with raised errors

FelipeCybis committed:
Added function to valided hashs from hash list
* Gets data directory containing `group-*` directories. Inside each
  group directory there should be a `hash_list.txt` or `data_hashes.txt`
  with the hash lists

FelipeCybis committed:
Improved docstring
* raised errors were added
* precise information on the `hash_list.txt` location was added

FelipeCybis committed:
Fixed sequence of instructions
* We would not be in the right directory for the commands, now we are.

FelipeCybis committed:
Merge branch 'main' into validate-data

FelipeCybis committed:
Merge remote-tracking branch 'upstream/main' into validate-data

FelipeCybis committed:
Enabled `validate_data` to receive multiple paths
* Enabling validation of different group directories directly

FelipeCybis committed:
Improved docstring to new logic after Matthew PR

FelipeCybis committed:
Getting the directory logic correct
* We need to get the parent of the input path to join with the filename
  taken from the ``hash_list.txt`` to get a path that exists

FelipeCybis committed:
Merge branch 'validate-data' of https://github.com/FelipeCybis/diagnostics-IcoTeam into validate-data

FelipeCybis committed:
Resolving merge conflicts

sdiebolt commented on scripts/validate_data.py:
```diff
@@ -21,54 +21,116 @@ def file_hash(filename):
-    groups = list(group_directory.glob('group-??'))
+    groups = list(group_directory.glob('group-02'))
     if len(groups) == 0:
         raise RuntimeError('No group directory in data directory: '
                            'have you downloaded and unpacked the data?')
 
-    if len(groups) > 1:
-        raise RuntimeError('Too many group directories in data directory')
+    # if len(groups) > 1:
+    #     raise RuntimeError('Too many group directories in data directory')
     # Call function to validate data in data directory
-    validate_data(groups[0])
+    validate_data(*groups)
```
Maybe we should keep the logic of the `main` function, since it's part of the project validation, and instead remove the `group-00` folder?

sdiebolt commented on scripts/validate_data.py:
```diff
@@ -21,54 +21,116 @@ def file_hash(filename):
+            # If hash for filename is not the same as the one in the file, raise
+            # ValueError
+            if not actual_hash == expected_hash:
+                raise ValueError(
```
We should maybe raise a `RuntimeError` here. `ValueError` are generally raised for unexpected values of arguments.

sdiebolt committed:
Delete unwanted hash list file

sdiebolt committed:
Restored `main` function 

The `main` functions was slightly altered to handle the `group-00` folder introduced by the last PR. Instead, `group-00` is now removed.
